src/mcp/client/auth.py

```diff
@@ -120,6 +121,42 @@ def _get_authorization_base_url(self, server_url: str) -> str:
         # Remove path component
         return urlunparse((parsed.scheme, parsed.netloc, "", "", "", ""))
 
+    async def _discover_protected_resource_metadata(self, server_url: str) -> ProtectedResourceMetadata | None:
+        """
+        Discover protected resource metadata from server's well-known endpoint.
+        RFC 9728 Protected Resource Metadata.
+        """
+        # Extract base URL per MCP spec
+        auth_base_url = self._get_authorization_base_url(server_url)
+        url = urljoin(auth_base_url, "/.well-known/oauth-protected-resource")
+        headers = {MCP_PROTOCOL_VERSION: LATEST_PROTOCOL_VERSION}
+
+        async with httpx.AsyncClient() as client:
+            try:
+                response = await client.get(url, headers=headers)
+                if response.status_code == 404:
+                    return None
+                response.raise_for_status()
+                metadata_json = response.json()
+                logger.debug(f"Protected resource metadata discovered: {metadata_json}")
+                return ProtectedResourceMetadata.model_validate(metadata_json)
```

Comment:
We don't need to convert to JSON first.
```suggestion
                metadata = ProtectedResourceMetadata.model_validate_json(response.content)
                logger.debug(f"Protected resource metadata discovered: {metadata}")
                return metadata
```